### PR TITLE
Automated Release Process

### PR DESCRIPTION
**Pull request by mgorsk1**
Signed-off-by: mgorsk1 <gorskimariusz13@gmail.com>

### CODE REVIEW

**Comment by Golodhros** [comment]
I totally support this! We have been talking internally about this for a while.
In my mind, we would have a @ci or @next release process and an official monthly release cadence, so we cover both people looking for only stable releases or user trying to get the latest bug fixes.

**Comment by mgorsk1** [comment]
Thanks @Golodhros , that's another thing I wanted to raise - intermediate releases (something in shape of release candidates). In ING to be as current as possible with latest changes in Amundsen now we rely on git submodules and building Amundsen components ourselves from maste, but having possibility to do `pip install --pre` would come definitely in handy.

File: rfcs/020-automated-release-process.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+- periodical releases every month
+- pre-releases with each change added to master
+
+## Reference-level Explanation (aka Technical Details)
+
+The plan for each release option is:
+
+### Continuous Pre-releases
+
+- Implement a new github action called `Pre-releases` that will trigger on every PR merged into master
```
Comment (mgorsk1) [inline rfcs/020-automated-release-process.md:34]: maybe to reduce no of pre-releases we could do them like monthly but with weekly granularity ?

**Comment by mgorsk1** [comment]
slightly off-topic but releasing alongside docker images could also be useful and made onboarding Amundsen even easier

File: rfcs/000-support-rdb-models.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
+- Feature Name: Support Relational Database Models
```
Comment (mgorsk1) [inline rfcs/000-support-rdb-models.md:1]: is it correct to have this submission under this PR ?

File: rfcs/020-automated-release-process.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+## Guide-level Explanation (aka Product Details)
+
+Having enforced naming convention for MRs we can leverage this to automatically categorize MRs and render them in
+changelog. By using automated github actions we can remove manual step from release process (releasing, rendering changelog)
+and replace it with three operations:
+
+- creating a git tag triggers a release
+- periodical releases every month
+- pre-releases with each change added to master
```
Comment (mgorsk1) [inline rfcs/020-automated-release-process.md:26]: my feeling is that'd be too granular (as explained in another comment)

File: rfcs/000-support-rdb-models.md (hunk 2 above)
Comment (Golodhros) [reply rfcs/000-support-rdb-models.md:1]: Oops, something went wrong!

File: rfcs/020-automated-release-process.md (hunk 3 above)
Comment (Golodhros) [reply rfcs/020-automated-release-process.md:26]: My thinking here is to cover this use case:

1. User reaches out about a bug
2. We fix it, merge it into master
3. We need to trigger a new release and ask user to update

Instead, we would just fix it and get it into master, asking the user to use the prerelease to get the latest

If all is automatic for us, I don't see any problem. Another thing is if we can't do it automatically, that would be tedious!

**Comment by feng-tao** [comment]
@allisonsuarez are we ok to merge this RFC given all the work you did is merged?

**Review by allisonsuarez — APPROVED** [review]

**Pull request merged**